### DISCUSSION TITLE
Boot Loop with ESP32

### DISCUSSION THREAD

**Discussion by d1One** [tttapa/Control-Surface#667] [Q&A]
Hi. 
I am trying the example VU-Meter_Exemple with an ESP32 DevKitC_V4 (SOLO-1) + SPI sd1306 and it seems I'm stuck in a boot loop, it resets every 6 seconds give or take. If I use the Adafruit Oled example everything works fine. I haven't changed anything in the VU-Meter_Exemple other than the oled pins. Any idea on what to look for?

```12:2:18.724 -> asks did not reset the watchdog in time:
12:2:18.724 -> E (11287) task_wdt: - IDLE0 (CPU 0)
12:2:18.724 -> E (11287) task_wdt: Tasks currently running:
12:2:18.724 -> E E (11287) task_wdt: Task watchdog got triggered. The following t(11287) task_wdt: CPU 0: loopTask
12:2:18.727 -> E (11287) task_wdt: Aborting.
12:2:18.727 -> abort() was called at PC 0x4015c2d4 on core 0
12:2:18.727 ->
12:2:18.727 -> ELF file SHA2c01:0x3ffbf3e0 0x4015c2d4:0x3ffbf400 0x4008d5ea:0x3ffbf420 0x40084115:0x3ffc73c0 0x40084da3:0x3ffc73e0 0x400817ef:0x3ffc7400 0x4x4008fbee:0x3ffc7480
12:2:18.757 ->
12:2:18.757 -> Rebooting...
12:2:18.757 -> ets Jun 8 2016 00:22:57
12:2:18.761 -> 56: 0000000000000000
12:2:18.761 ->
12:2:18.761 -> Backtrace: 0x4008e99b:0x3ffbf3c0 0x4008e00d2bf4:0x3ffc7420 0x400d1a4c:0x3ffc7440 0x400d7aa4:0x3ffc7460 0
12:2:18.761 -> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
12:2:18.761 -> configx00,hd_drv:0x00,wp_drv:0x00
12:2:18.789 -> mode:DIO, clock div:2
12:2:18.789 -> load:0x3fff0x40078000,len:10944
12:2:18.789 -> load:0x40080400,len:6360
12:2:18.789 -> entry 0x400806b4 sip: 0, SPIWP:0xee
12:2:18.792 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0018,len:4
12:2:18.792 -> load:0x3fff001c,len:1216
12:2:18.792 -> ho 0 tail 12 room 4
12:2:18.792 -> load:0
12:2:24.728 -> configsip: 0, SPIWP:0xee
12:2:24.728 -> clk_drv:0x00,q_drv:0x07c12a:0x3ffc73a0 0x400d3c4a:0x3ffc73c0 0x400d3cc2:0x3ffc73e0 0x4x400d7aa4:0x3ffc7460 0x4008fbee:0x3ffc7480
12:2:24.732 ->
12:2:24.732 -> Rebooting...
12:2:24.732 -> ets Jun 8 2016 00:22:57
12:2:24.732 ->
12:2:24.732 -> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FA0,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
12:2:24.761 -> mode:DIO, clo360
12:2:24.761 -> entry 0x400806b4
12:2:24.765 -> ck div:2
12:2:24.765 -> load:0x3fff0018,len:4
12:2:24.765 -> load:0x3fff001c,len:1216
12:2:24.765 -> ho 0 tail 12 room 4
12:2:24.765 -> load:0x40078000,len:10944
12:2:24.765 -> load:0x40080400,len:6asks did not reset the watchdog in time:
12:2:30.664 -> E (11271) task_wdt: - IDLE0 (CPU 0)
12:2:30.664 -> E (11271) task_wdt: Tasks currently running:
12:2:30.664 -> E E (11271) task_wdt: Task watchdog got triggered. The following t(11271) task_wdt: CPU 0: loopTask
12:2:30.666 -> E (11271) task_wdt: Aborting.
12:2:30.696 -> abort() was called at PC 0x4015c2d4 on core 0
12:2:30.696 ->
12:2:30.696 -> ELF file SHA256: 0000000000000000
12:2:30.696 ->
12:2:30.696 -> Backtrace: 0x4008e99b:0x3ffbf3c0 0x4008ed2bf4:0x3ffc7420 0x400d1a4c:0x3ffc7440 0x400d7aa4:0x3ffc7460 0x4c01:0x3ffbf3e0 0x4015c2d4:0x3ffbf400 0x4008d5ea:0x3ffbf420 0x400008fbee:0x3ffc7480
12:2:30.699 ->
12:2:30.699 -> Rebooting...
```

**Comment by tttapa** [comment]
Related: #646

You need the master version of Control Surface if you're using a recent version of the ESP32 Arduino core.

**Reply by d1One** [reply]
> Related: #646
> 
> You need the master version of Control Surface if you're using a recent version of the ESP32 Arduino core.

Thanks very much for the reply! Assuming that's done by downloading the zip version in the green button instead of 1.2.0 release that's what I am using, I followed the install instructions on the documentation.

Could this be because my ESP32 is not officially supported by Arduino Core? It's single-core and Control-Surface is built for multi-core? I am using this to force my board to be able to boot multi-core sketches:
https://github.com/lbernstone/arduino-esp32-solo It's only available up to Arduino 1.0.6

If I don't I get:
```
22:36:46.103 -> E (310) cpu_start: Running on single core variant of a chip, but app is built with multi-core support.
22:36:46.103 -> E (310) cpu_start: Check that CONFIG_FREERTOS_UNICORE is enabled in menuconfig
22:36:46.135 ->
22:36:46.135 -> abort() was called at PC 0x40082db9 on core 0
```

I've ordered another board with dual-core, hopefully, that one works. Thanks for your time.

**Reply by tttapa** [reply]
I've never tried a single-core ESP32, the code was written with a dual-core CPU in mind. In the default configuration, the Bluetooth stack runs on core 0, and the normal Arduino code runs on core 1.

However, although not optimal for performance, the code should still run fine when there's only one core available. Could you decode the backtrace you get using https://github.com/me-no-dev/EspExceptionDecoder?  
If you add print statements to the setup and loop, can you see where it crashes? Does it get to the setup? Does it crash before, during or after  `Control_Surface.begin()`?

**Reply by d1One** [reply]
I've tried installing the EspExceptionDecoder but it's not showing up under the Tools menu.

I just received a new dual-core board and everything works fine, the combo of having to use a hacky approach to force the sketch upload + old Arduino core version + being a slower board overall isn't worth the effort tbh... Thanks for getting back at me tho, appreciate it.
